### DISCUSSION TITLE
posts/2023/08/why-style-link-as-button/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#31] [Announcements]
# posts/2023/08/why-style-link-as-button/

Fixing a long-standing sin against accessibility — and, as it turns out, against correct HTML.

https://www.brycewray.com/posts/2023/08/why-style-link-as-button/

**Comment by davidsneighbour** [comment]
The next error is having a `div` (block level) element inside of an `a` (inline level) element. You want to change that to `span` and then, to "imitate" block level behaviour, add `display: inline-block;` to the `span`. With the proper CSS you could also get rid of the wrapper for the label itself completely, but it might lead to issues with vertical align in it's current state.

**Comment by brycewray** [comment]
> The next error is having a `div` (block level) element inside of an `a` (inline level) element.

Apparently [allowed within HTML 5](https://stackoverflow.com/questions/1827965/is-putting-a-div-inside-an-anchor-ever-correct), although I’ll keep looking into this.

**Update**: Post and actual code (HTML and CSS) now edited accordingly.
